radarlabs/radar-sdk-ios
# Add rto tracking options update to trackOnce and trackVerified calls

We previously only updated the RTO settings with background tracking calls, this PR adds a side-effect to the `trackOnce` and `trackVerified` calls such that any updates to RTO tracking options are saved into settings.


QA:
steps: 

1. Have RTO turned off
2. check that RTO is not on in waypoint setting: 
<img width="232" alt="image" src="https://github.com/radarlabs/radar-sdk-ios/assets/139801512/604fa22a-e859-4668-8d85-415701ff8f78">

3. turn on RTO to efficient in dashboard
5. call track once in waypoint and verify that it updated
<img width="433" alt="image" src="https://github.com/radarlabs/radar-sdk-ios/assets/139801512/55e152d9-99c4-486d-80f9-e7e3db265476">


6. modify RTO settings on dashboard and verify that it is propagated to waypoint after a calling trackOnce 
<img width="435" alt="image" src="https://github.com/radarlabs/radar-sdk-ios/assets/139801512/b836a848-09fa-4f21-b0b7-2ff81cbe845e">


8. turn off RTO on dashboard and call another trackonce on waypoint to verify that RTO is turned off
<img width="399" alt="image" src="https://github.com/radarlabs/radar-sdk-ios/assets/139801512/f4cc0277-79b0-4fab-bbff-e06c9220fcba">

10. repeat the previosu steps but this time use the trackverify call


Follow up question, do we wish to introduce this side effect to the track verified calls and the mock tracking calls too?
